src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -612,15 +631,32 @@ def acr_task_run(cmd,
 
     from ._client_factory import cf_acr_registries
     client_registries = cf_acr_registries(cmd.cli_ctx)
-    TaskRunRequest = cmd.get_models('TaskRunRequest')
+
+    #TaskRunRequest = cmd.get_models('TaskRunRequest')
+    from .sdk.models import TaskRunRequest, OverrideTaskStepProperties
+
+    import base64
+    if update_trigger_token:
+        update_trigger_token = base64.b64encode(update_trigger_token.encode()).decode()
+
+    task_id=get_task_id_from_task_name(cmd.cli_ctx, resource_group_name, registry_name, task_name)
+
+    override_task_step_properties = OverrideTaskStepProperties(
+        context_path=context,
+        file=file,
+        arguments=(arguments if arguments else []) + (secret_arguments if secret_arguments else []),
```

Comment:
>arguments [](start = 8, length = 9)

Do we merge argument and secret_arguments in the override case?

Reply:
Yes, we always merge `arguments/secret_arguments` and `setValue/setSecret`. Same for override